chroot_setup

```diff
@@ -0,0 +1,440 @@
+  install_template = Template('/QOpenSys/pkgs/bin/yum $yes --installroot=$chroot_dir install $package')
+  auto_yes = '-y' if args.auto_yes else ''
+  for package in package_list:
+    command = install_template.substitute(yes=auto_yes, chroot_dir=args.chroot_directory, package=package)
+    logging.debug(command)
+    os.system(command)
+
+#
+# MAIN
+#
+if __name__ == '__main__':
+  lst = 0
+  ops = 0
+  qsys = 0
+  CHROOT_DIR = ""
+  CHROOT_LIST = ""
+  SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__)) + "/config"
+
+  print(SCRIPT_DIR)
+
+  auto_yes = args.auto_yes
+  initial_check()
+
+  # This is done because the arg may be a relative path name.
+  # Need to be done before any 'cd'
+  TEMP_DIR = os.path.realpath(args.chroot_directory)
+  print(TEMP_DIR)
+
+  # if -v was passed, need to turn logging to DEBUG so all logging messages in the
+  # code are printed
+  if (args.verbose):
+    logging.basicConfig(level=logging.DEBUG)
+
+  logging.debug("Script directory: " + SCRIPT_DIR)
+  os.chdir(SCRIPT_DIR)
+  logging.debug("PWD: " + os.getcwd())
+
+  if not TEMP_DIR.startswith('/QOpenSys/'):
+    print("\nERROR: [CHROOT DIRECTORY] Must begin with /QOpenSys/...\n")
+    parser.print_help
+    sys.exit(-2)
+
+  CHROOT_DIR = TEMP_DIR
+
+  if not os.path.isdir(CHROOT_DIR):
+    print(CHROOT_DIR + " does not exist.")
+    logging.debug(f"Creating directory at {CHROOT_DIR}")
+    os.mkdir(CHROOT_DIR)
+  else:
+    if not auto_yes:
+      print(CHROOT_DIR + " directory already exists")
+      val = input("Would you like to continue chroot setup? [y/N]")
+      if not val == "y":
+        print("Bye")
+        sys.exit(-4)
+
+
+  # Validate CHROOT TYPE Argument(s)
+  if not args.chroot_type:
+    # Only chroot_directory was provided to the script:
+    # Ask if minimal with includes chroot is desired
+    if not auto_yes:
+      val = input("Would you like to create a minimal chroot w/ includes @ " + CHROOT_DIR + "? [y/N]:")
+      if not val == "y":
+        print("Specify your desired [CHROOT TYPE]")
+        parser.print_help
+        sys.exit(-5)
+
+    mylist = []
+    mylist.append(SCRIPT_DIR + "/chroot_includes.lst")
+    mylist.append(SCRIPT_DIR + "/chroot_minimal.lst")
+
+  else:
+    # If chroot types were specified, then use those
+    for arg in args.chroot_type:
+      print("Chroot type is: " + arg)
+      
+      CHROOT_LIST = arg
+
+      # add chroot prefix if it doesn't exist
+      if not CHROOT_LIST.startswith('chroot'):
+        CHROOT_LIST = 'chroot_' + CHROOT_LIST
+
+      # check that the .lst file name was given
+      if not CHROOT_LIST.endswith('.lst'):
+        CHROOT_LIST = CHROOT_LIST + '.lst'
+
+      # check that the .lst file exists
+      if not os.path.isfile(CHROOT_LIST):
+        print(f"{CHROOT_LIST} cannot be found in {os.cwd}")
+        sys.exit(-7)
+      else:
+        # append file name to end of list
+        mylist.append(os.readlink(CHROOT_LIST))
```

Comment:
Probably should open them too, while we're at it. If we still want to print the file path later, we could store a tuple(path, file obj) in the array.

Reply:
Fixed up in d80220a

Instead of checking if the file exists I just append the file name to the last. later in the chroot_setup function the file is opened for reading.